### DISCUSSION TITLE
Can we take Database table as input in any way ?

### DISCUSSION THREAD

**Discussion by Anto1106** [imartinez/privateGPT#595] [Q&A]
Hello I have a database with me with connection string can we take that as a input and analyse the data?

**Comment by sime2408** [comment]
[langchain sql docs](https://python.langchain.com/en/latest/modules/chains/examples/sqlite.html)

**Comment by ghostersk** [comment]
Bump this question up.
Is there some way how to link sql database to PrivateGPT so it can use it instead of files?
I would like to link it to BookStack Knowledge Base, that is stored in sql not in files.

The link above is 404, and i do not see PrivateGPT on that langchain project.

**Comment by tlinnet** [comment]
Privategpt is build around documents as input. There is two ways. Either you make a script that export the sql rows as documents and import these.

Or you you make a PR that implement reading SQL into Privategpt. And handle drivers. And authentication. And selection. And UI integration.
